seed/data_importer/tasks.py

```diff
@@ -785,14 +786,15 @@ def _save_greenbutton_data_task(readings, meter_id, meter_usage_point_id, progre
             )
             with connection.cursor() as cursor:
                 cursor.execute(sql)
-                result['source_id'] = meter_usage_point_id
-                result['count'] = len(cursor.fetchall())
+                key = "{} - {}".format(meter_usage_point_id, meter.get_type_display())
+                result[key] = {'count': len(cursor.fetchall())}
     except ProgrammingError as e:
         if "ON CONFLICT DO UPDATE command cannot affect row a second time" in str(e):
-            result['source_id'] = meter_usage_point_id
-            result['error'] = "Overlapping readings."
-    except Exception:
-        return progress_data.finish_with_error('data failed to import')
+            key = "{} - {}".format(meter_usage_point_id, meter.get_type_display())
+            result[key] = {"error": "Overlapping readings."}
+    except Exception as e:
```

Comment:
@axelstudios Here and L853 are the small changes I made to improve error handling on Meter Import